queue_consumer.go

```diff
@@ -132,9 +137,18 @@ func (mf *Consumer) receiveMessages(ctx context.Context, wg *sync.WaitGroup, don
 // use the WithShutdownChan RunOption.
 //
 // If the context is canceled, the returned error is the context's error.
+// If the optional shutDown channel is closed before Run is called, the returned error is ErrShutdownChannelClosed.
 // If in-flight messages drain to completion after shutdown, the returned error is nil.
 func (mf *Consumer) Run(ctx context.Context, opts ...RunOption) error {
-	runOptions := resolveRunOptions(opts)
+	ro := resolveRunOptions(opts)
+	select {
+	case _, open := <-ro.shutDown:
+		if !open {
```

Comment:
You don't need to check the result if `ro.shutDown` is a close-only channel (the convention for "done" channels). It can simply be `case <- ro.shutDown: return ErrShutdownChannelClosed`.

Reply:
Sounds good.  Will fix.